src/main/java/com/parsing/parsers/pdf/parsing/DataRecognizer.java

```diff
@@ -0,0 +1,156 @@
+        if (totalPrice != null && totalPrice.equals(BigDecimal.valueOf(amount).multiply(price))) {
+            lotPDFResultService.saveLaptopItem(model, price, amount);
+            return true;
+        }
+        return false;
+    }
+
+    private void getAmount(List<String> amounts) {
+        if (amounts.size() == 1) {
+            amount = Integer.parseInt(amounts.get(0));
+        } else if (amounts.size() > 1) {
+            log.debug("more than one amount number found in one row");
+            amount = 0;
+        }
+        if (amount == 0 && price != null) {
+            amount = totalPrice.divide(price).toBigInteger().intValueExact();
+        }
+    }
+
+    private void getPriceAndTotalPrice(List<String> prices) {
+        List<BigDecimal> bigDecimals = null;
+        try {
+            bigDecimals = prices.stream().map(BigDecimal::new).toList();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        Optional<BigDecimal> min = bigDecimals.stream().min(Comparator.naturalOrder());
+        Optional<BigDecimal> max = bigDecimals.stream().max(Comparator.naturalOrder());
+        if (min.isPresent() && max.isPresent()) {
+            price = min.get();
+            totalPrice = max.get();
+        }
+    }
+
+    private int getModelColumnNumber(Row row) {
+        int modelColumnNumber = 0;
+        for (Column column : row.getColumns()) {
+            if (isModel(column.getParsingResult())) {
+                model = findModel(column.getParsingResult());
+                modelColumnNumber = row.getColumns().indexOf(column);
+            }
+        }
+        return modelColumnNumber;
+    }
+
+    private boolean isModelRow(Row row) {
+        List<String> listLowerCaseColumn = row.getColumns().stream().map(x -> x.getParsingResult().toLowerCase(Locale.ROOT)).toList();
+        for (String column : listLowerCaseColumn) {
+            for (String laptopModel : laptopModels) {
+                if (column.contains(laptopModel)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private String findModel(String input) {
+        for (String laptop : laptopModels) {
+            if (input.toLowerCase(Locale.ROOT).contains(laptop)) {
+                return input;
+            }
+        }
+        return null;
+    }
+
+    private boolean isModel(String input) {
+        for (String laptop : laptopModels) {
+            if (input.toLowerCase(Locale.ROOT).contains(laptop)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String findPrices(String input) {
+        String result = null;
+        if (input.contains(",")) {
+            result = input.replace(" ", "").replace("\n", "").replace(",", ".").replace("І", "1");
+        }
+        return result;
+    }
+
+    private String findAmount(String input) {
+        if (input.length() < 4 && input.length() > 1) {
```

Comment:
4 is a magic number, it should be something like MIN_PRICE_AMOUNT_DIGITS

Reply:
Done.